### PR TITLE
Move Windows builds to Github Actions

### PR DESCRIPTION
**Pull request by ncesario-lunarg**
This PR does a few things (in commit order):

- Add a test to verify cmake's minimum version still works
- Moves the CI builds from AppVeyor to github actions (thanks to Lenny's work in Vulkan-Loader!)
- Refactors some more "common CI" code into common_ci.py

### CODE REVIEW

**Comment by ncesario-lunarg** [comment]
This also appears to fixe clang builds. It looks like before the clang builds were still [using gcc](https://github.com/KhronosGroup/Vulkan-ValidationLayers/pull/2766/checks?check_run_id=2494765281#step:4:777).

There is some weirdness with the format checking I need to figure out that I wasn't seeing before making a PR.

**Comment by ncesario-lunarg** [comment]
Sorry for the churn with this. There's still some difficulty with the format check. I'm trying to use the v2 checkout which has some desirable effects, but may end up being more trouble than its worth (w.r.t. to the format check). Marking this as a draft until this issue is resolved (unfortunately this only seems to be a problem with PRs, so I need to keep pushing to this branch...).

**Review by mikes-lunarg — COMMENTED** [review]
I'm surprised to see the new file `github_ci_win.py`. I would expect `github_ci_win_linux.py` to work on windows with minimal tweaking, that also seems to have been @mark-lunarg's original intent based on the script name. If `github_ci_win_linux.py` really did work on both platforms that would eliminate the need for some of the refactoring here.

File: scripts/cmake_test_min.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+# Test to make sure cmake is compliant with minimum version
+# NOTE: this assumes VVL has already been "setup" (update_deps.py has been run, etc.)
+
+import os, platform, subprocess
+from urllib.parse import urlparse
+import utils.utils as utils
+
+curr_platform = platform.system().lower()
+cmake_version = '3.10.2'
```
[the remark this answers is not in the document]
Comment (ncesario-lunarg) [reply scripts/cmake_test_min.py:27]: Good idea! I don't think that should be too difficult; will give it a try.

**Comment by ncesario-lunarg** [comment]
> I'm surprised to see the new file `github_ci_win.py`. I would expect `github_ci_win_linux.py` to work on windows with minimal tweaking, that also seems to have been @mark-lunarg's original intent based on the script name. If `github_ci_win_linux.py` really did work on both platforms that would eliminate the need for some of the refactoring here.

I agree, and this is what I spent a bit of time on initially. Unfortunately, it doesn't seem trivial to get everything building on Windows (at least I had difficulty on my machine and whatever is running on GH actions), and I later found out (when I _actually looked_ at the details of the .appveyor file...which I should have done _first_) the AppVeyor build does not call the `github_ci_win_linux.py` script. I maybe should rename `github_ci_win_linux.py` to `github_ci_linux.py`, but currently left as-is as the intention is to _eventually_ get that script working on Windows as well.

**Comment by mikes-lunarg** [comment]
> I maybe should rename `github_ci_win_linux.py` to `github_ci_linux.py`, but currently left as-is as the intention is to _eventually_ get that script working on Windows as well.

That seems like a good plan. FWIW I was just able to hack up the current master version of `github_ci_win_linux.py` to build and run tests on Windows, so it isn't that far off. Getting it working in the Windows cloud environment is probably yet another hurdle.

**Review by mikes-lunarg — APPROVED** [review]

**Comment by ncesario-lunarg** [comment]
I created #2784 to track getting the `scripts/github_ci_win_linux.py` working on Windows. I'll try to get this in sooner rather than later.

File: scripts/utils/utils.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+# Various utility functions/classes
+
+import os, shutil
+import urllib.parse
+from urllib.parse import urlparse
+import http.client as http
+
+def make_or_exist_dirs(path, clean=False):
+    "Wrapper for os.makedirs that tolerates the directory already existing"
+    # Could use os.makedirs(path, exist_ok=True) if we drop python2
```
[the remark this answers is not in the document]
Comment (ncesario-lunarg) [reply scripts/utils/utils.py]: I voted for just making this `make_dirs`. The original make_or_exist_dirs was left unchanged. I also went with the "python-3-only" option as I'm pretty sure these scripts will not run on python 2 anymore. I'll go through our documentation and make sure the references to python 2 are removed in a separate PR.

**Comment by ncesario-lunarg** [comment]
Again, sorry for the churn. Lenny helped me get much further with the Windows CI, but it's still not quite there in terms of getting the tests running. It's close, but I'd like to get this in before the "last 10%" sucks up any more time.
